src/ngraph/runtime/aligned_buffer.cpp

```diff
@@ -14,33 +14,40 @@
 // limitations under the License.
 //*****************************************************************************
 
+#include <iostream>
 #include <memory>
 
 #include "ngraph/runtime/aligned_buffer.hpp"
+#include "ngraph/runtime/allocator.hpp"
 #include "ngraph/util.hpp"
 
 using namespace ngraph;
+using namespace std;
 
 runtime::AlignedBuffer::AlignedBuffer()
-    : m_allocated_buffer(nullptr)
+    : m_allocator(nullptr)
+    , m_allocated_buffer(nullptr)
     , m_aligned_buffer(nullptr)
     , m_byte_size(0)
 {
 }
 
-runtime::AlignedBuffer::AlignedBuffer(size_t byte_size, size_t alignment)
+runtime::AlignedBuffer::AlignedBuffer(size_t byte_size, size_t alignment, Allocator* allocator)
+    : m_allocator(allocator)
+    , m_byte_size(byte_size)
 {
-    m_byte_size = byte_size;
     if (m_byte_size > 0)
     {
-        size_t allocation_size = m_byte_size + alignment;
-        m_allocated_buffer = static_cast<char*>(ngraph_malloc(allocation_size));
+        m_allocated_buffer = static_cast<char*>(m_allocator->Malloc(m_byte_size, alignment));
```

Comment:
I know that I asked for this change but I now think that this will be easier if we revert to your previous code where you defaulted the Allocator to a nullptr and used the regular old AlignedBuffer calculations for aligning memory. It is the most portable.